astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -272,8 +272,7 @@ def hcrs_to_hcrs(from_coo, to_frame):
 @frame_transform_graph.transform(FunctionTransformWithFiniteDifference, TETE, TETE)
 def tete_to_tete(from_coo, to_frame):
     if (np.all(from_coo.obstime == to_frame.obstime) and
```

Comment:
Can we swap these round? A common use pattern is for location to be singular and obstime to be a large array. If the conditions are reversed then the faster condition may fail first and avoid the longer check on obstime.

Reply:
Yes, makes sense.